### PR TITLE
feat: avalanche support for native wallet

### PR DESCRIPTION
**Pull request by chxash**
This adds avalanche support for native wallet by using `_supportsEthSwitchChain` boolean.
Tested with web, I have been able to send receive `AVAX`, `WETH on Avalanche` and perform swaps `AVAX` <=> `WETH`, see screenshots attached.

avax_sent_from_native

weth_sent_native

avax_trade_native_1

avax_trade_native_2

avax_trade_native_3

avax_trade_native_4

avax_tx_history

**Issue**
closes https://github.com/shapeshift/hdwallet/issues/580

### CODE REVIEW

File: packages/hdwallet-native/src/ethereum.ts [hunk 1]
```diff
@@ -27,6 +28,15 @@ export function MixinNativeETHWalletInfo<TBase extends core.Constructor<core.HDW
       return true;
     }
 
+    async ethGetChainId(): Promise<number | null> {
```
Comment (chxash) [inline packages/hdwallet-native/src/ethereum.ts:31]: I am unsure about the necessity of doing this (enables the switch button on web, but has not real utility currently)

File: packages/hdwallet-native/src/native.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ export class NativeHDWallet
   readonly _supportsBTC = true;
   readonly _supportsETH = true;
   readonly _supportsCosmos = true;
-  readonly _supportsEthSwitchChain = false;
+  readonly _supportsEthSwitchChain = true;
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply packages/hdwallet-native/src/native.ts:217]: Note, we will also need to add it in all wallets that currently support `_supportsEthSwitchChain` (in addition to `_supportsEthSwitchChain` itself)

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
`slip44Table` addition looks sane and really is all we should need without the `ethGetChainId` and `ethSwitchChain` methods, which are effectively wrappers around [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193), the standard JSON-RPC provider API used by MetaMask and MetaMask lookalikes.

We will need a new feature detection flag for native, `_supportsEthSwitchChain` unfortunately won't work for us: it is intricately tied to some web logic to do chain switching / display chain-switching abilities as a dropdown, and native doesn't and shouldn't have any notion of a chain switch

While we initially went with `_supportsAvalanche`, the decision was then made to use chain switching as a feature detection instead (https://github.com/shapeshift/hdwallet/pull/548) which unfortunately your PR happens to unveil was a suboptimal choice - we will need both chain switching detection **and** avalanche detection

File: packages/hdwallet-native/src/native.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (chxash) [reply packages/hdwallet-native/src/native.ts]: done !

**Comment by chxash** [comment]
> `slip44Table` addition looks sane and really is all we should need without the `ethGetChainId` and `ethSwitchChain` methods, which are effectively wrappers around [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193), the standard JSON-RPC provider API used by MetaMask and MetaMask lookalikes.
> 
> We will need a new feature detection flag for native, `_supportsEthSwitchChain` unfortunately won't work for us: it is intricately tied to some web logic to do chain switching / display chain-switching abilities as a dropdown, and native doesn't and shouldn't have any notion of a chain switch
> 
> While we initially went with `_supportsAvalanche`, the decision was then made to use chain switching as a feature detection instead (#548) which unfortunately your PR happens to unveil was a suboptimal choice - we will need both chain switching detection **and** avalanche detection

Fixed all of this, thanks !
Tested in web by using hdwallet locally with branch https://github.com/shapeshift/web/tree/feat_avalanche_support_native

Tested sending / receiving avax with native and metamask wallets. Swaps WETH on avalanche <> AVAX work fine.
Switch chain button is no more visible for native wallet.

native1_avax

metamask_avax

**Review by gomesalexandre — DISMISSED** [review]
LGTM, tested locally and I was able to display my AVAX receive address, and after sending to it, the balance properly updates and the Tx shows up in the Tx history as a receive:

image
image

Pre-stamping this, but before this gets merged, can you run `yarn lerna:version-minor` and push the commit so this autopublishes? 🙏🏽

**Review by gomesalexandre — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**